HW_Task/Task_1.py

```diff
@@ -0,0 +1,76 @@
+'''
+Задание
+Напишите следующие функции:
+1. Нахождение корней квадратного уравнения.
+2. Генерация csv файла с тремя случайными числами в каждой строке 100-1000 строк.
+3. Декоратор, запускающий функцию нахождения корней квадратного уравнения с каждой тройкой чисел из csv файла.
+4. Декоратор, сохраняющий переданные параметры и результаты работы функции в json файл.
+'''
+
+import csv
+import datetime
+import json
+import math
+import os.path
+from random import randint as rnd
+from typing import Callable
+
+
+def deco_csv(function: Callable):
+    create_coefficients_csvfile()
+
+    def wrapper():
+        with open('coefficiens.csv', 'r', encoding='UTF-8') as file:
+            data = csv.reader(file, quoting=csv.QUOTE_NONNUMERIC)
+            for coefficient in data:
+                if coefficient and coefficient[0] != 0:
+                    function(*coefficient)
+
+    return wrapper
+
+
+def json_result(func: Callable):
+    result = {}
+    if os.path.exists('solutions.json'):
+        with open('solutions.json', 'r', encoding='UTF-8') as file:
+            result = json.load(file)
+    else:
+        with open('solutions.json', 'w', encoding='UTF-8') as file:
+            json.dump(result, file)
```

Comment:
Не совсем понятно, зачем записывать пустой словарь, когда ниже уже будет записан словарь, содержащий какие-то данные. Но вот замкнутую переменную создать можно
```suggestion
    if os.path.exists('solutions.json'):
        with open('solutions.json', 'r', encoding='UTF-8') as file:
            result = json.load(file)
    else:
        result = {}
```